src/h5web/providers/models.ts

```diff
@@ -60,3 +62,5 @@ export type Value<
   S extends HDF5Shape,
   T extends HDF5Type
 > = S extends HDF5SimpleShape ? PrimitiveType<T>[] : PrimitiveType<T>;
+
+export type ComplexArray = (ComplexArray | Complex)[];
```

Comment:
With my comment above, I think this will type no longer be relevant. It should be achieved with type `Value` above.